packages/color-scale-block/src/types.ts

```diff
@@ -0,0 +1,121 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { MouseEvent } from 'react';
+import { Color, ColorFormat, DraggableItem, DropZonePosition, Palette } from '@frontify/fondue';
+
+export type Settings = {
+    customHeight: boolean;
+    heightSlider: string;
+    heightInput: string;
+    ['color-input']: ColorProps[];
+};
+
+export type FormattedColor = {
+    id?: number | null;
+    alpha?: number | undefined | null;
+    red?: number | null;
+    green?: number | null;
+    blue?: number | null;
+    name?: string | undefined | null;
+};
+
+export type ColorProps = {
+    id?: number;
+    sort?: number;
+    color?: Color;
+    width?: number;
+    alt?: string;
+};
+
+export type ColorPalette = {
+    id?: number;
+    title?: string;
+    source?: string;
+    colors?: FormattedColor[];
+};
+
+export type ColorScaleBlockRef = {
+    current: HTMLDivElement | null;
+};
+
+export type SquareWithColorProps = {
+    id: number | undefined;
+    index: number;
+    width: number;
+    height: string;
+    isFirst: boolean;
+    isLast: boolean;
+    currentColor: ColorProps;
+    calculateLeftPosition: (index: number, width: number) => void;
+    isEditing: boolean;
+    editedColor?: ColorProps | null | undefined;
+    setEditedColor: (color: ColorProps) => void;
+    updateColor: (color: ColorProps) => void;
+    setFormat: (color?: ColorFormat) => void;
+    deleteColor: (color: number | undefined) => void;
+    handleDrop: (targetItem: ColorProps, sourceItem: ColorProps, position: DropZonePosition) => void;
+    listId: string;
+    backgroundColorRgba?: string;
+    onResizeStart?: (event: MouseEvent, index?: number, currentColor?: ColorProps) => void | undefined;
+    totalNumberOfBlocks: number;
+    isDragging: boolean;
+    setCurrentlyDraggedColorId: (value: number | null | undefined) => void;
```

Comment:
`value?: number | null`

Reply:
Removed the undefined definition from setCurrentlyDraggedColorId and updated code accordignly.